core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -134,15 +140,15 @@ protected int run() throws Exception {
             a = new ParametersAction(values);
         }
 
-        if (checkSCM) {
-            if (job.poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
+        if (checkSCM && job instanceof AbstractProject<?, ?>) {
+            if (((AbstractProject<?, ?>)job).poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
```

Comment:
For example

```java
if (checkSCM) {
    SCMTriggerItem pollable = SCMTriggerItems.asSCMTriggerItem(job);
    if (pollable != null && pollable.poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
```

(I suspect the original code ought to have been using `hasChanges()` to deal with `INSIGNIFICANT`, but anyway that is outside the scope of this PR.)